lib/rentlinx/modules/linkable.rb

```diff
@@ -0,0 +1,38 @@
+module Rentlinx
+  module Linkable
+    def post_with_links
+      post
+      post_links
+    end
+
+    def post_links
+      Rentlinx.client.post_links(@links)
+    end
+
+    def links
+      return @links if @links
```

Comment:
You could make that argument.  Using <code>||=</code> is a very common ruby pattern for this kind of thing though:  http://www.justinweiss.com/blog/2014/07/28/4-simple-memoization-patterns-in-ruby-and-one-gem/.  It doesn't matter that much to me, though.